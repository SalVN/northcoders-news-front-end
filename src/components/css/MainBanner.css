.hero-css {
    background-color: #c1002e;
    min-height: 150px;
}

.main-heading {
    color: white !important;
    font-size: 40px;
}