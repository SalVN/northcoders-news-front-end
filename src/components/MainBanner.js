import React from 'react';
// import PropTypes from 'prop-types';
import './css/MainBanner.css';
import { BrowserRouter as Router, Link } from 'react-router-dom';

class MainBanner extends React.Component {
  render() {
    return (
      <section className='hero is-primary is-bold'>
        <div className='hero-body hero-css'>
          <div className='container'>
            <Link to='/'>
              <h1 className='title main-heading is-size-1 is-size-3-mobile'><strong>NORTHCODERS News</strong></h1>
            </Link>
          </div>
        </div>
      </section>
    );
  }
}

MainBanner.propTypes = {
};

export default MainBanner;