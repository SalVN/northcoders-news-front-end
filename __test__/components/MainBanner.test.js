import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';


import MainBanner from '../../src/components/MainBanner';

describe('MainBanner', () => {
    it('is a function', () => {
        expect(typeof MainBanner).toEqual('function');
    });

    it('renders', () => {
        const enzymeWrapper = shallow(<MainBanner />);
        expect(enzymeWrapper.children().length).toEqual(1);
    });

    it('contains the title NORTHCODERS News', () => {
        const enzymeWrapper = shallow(<MainBanner />);
        expect(enzymeWrapper.find('h1').nodes[0].props.children.props.children).toBe('NORTHCODERS News');
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <MainBanner />
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});