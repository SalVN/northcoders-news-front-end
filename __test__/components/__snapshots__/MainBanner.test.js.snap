// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainBanner renders correctly 1`] = `
<section
  className="hero is-primary is-bold"
>
  <div
    className="hero-body hero-css"
  >
    <div
      className="container"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h1
          className="title main-heading is-size-1 is-size-3-mobile"
        >
          <strong>
            NORTHCODERS News
          </strong>
        </h1>
      </a>
    </div>
  </div>
</section>
`;
